src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/service/cessations/BeneficialOwnerCessationService.java

```diff
@@ -0,0 +1,111 @@
+
+@Service
+public class BeneficialOwnerCessationService {
+
+  public static final String NO_PAIR_FOUND = "No pair found for the provided hashedId";
+
+  public List<Cessation> beneficialOwnerCessations(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+
+    List<Cessation> cessations = new ArrayList<>();
+    cessations.addAll(returnIndividualBeneficialOwners(overseasEntitySubmissionDto, combinedBoData));
+    cessations.addAll(returnCorporateEntityBeneficialOwnerCessations(overseasEntitySubmissionDto, combinedBoData));
+    cessations.addAll(returnLegalPersonBeneficialOwners(overseasEntitySubmissionDto, combinedBoData));
+    return cessations;
+  }
+
+  private List<IndividualBeneficialOwnerCessation> returnIndividualBeneficialOwners(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var beneficialOwnersIndividual = overseasEntitySubmissionDto.getBeneficialOwnersIndividual();
+    return beneficialOwnersIndividual.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnIndividualBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private List<CorporateEntityBeneficialOwnerCessation> returnCorporateEntityBeneficialOwnerCessations(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var returned = overseasEntitySubmissionDto.getBeneficialOwnersCorporate();
+    return returned.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnCorporateEntityBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private List<LegalPersonBeneficialOwnerCessation> returnLegalPersonBeneficialOwners(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var returned = overseasEntitySubmissionDto.getBeneficialOwnersGovernmentOrPublicAuthority();
+    return returned.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnLegalPersonBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private IndividualBeneficialOwnerCessation returnIndividualBeneficialOwnerCessation(
```

Comment:
A lot of repeated logic in these 3 methods- is it possible to extract some of it out?

Reply:
I had thought about that initially but because we're dealing with streams returning different objects each time it could make the code more unreadable by breaking it out. A fair spot but for readability sake I'd keep it.

Reply:
Agreed there is repeated logic, and I had went about refactoring it before but the code ended up less readable/maintainable unfortunately.